frontend/src/redux/reducers/ui.tsx

```diff
@@ -206,6 +211,47 @@ function reducer(state = _.cloneDeep(INITIAL_STATE), action: Action) {
       newFilters.branding.logo = component;
       break;
     }
+    case UI_SET_NOTIFICATIONS: {
+      const notifications = action.notifications;
+      /* There are two ways user can send notifications either a complete set of array or a single notification
+         when handling the array notifications we want to only have unique set of notifications pushed into the UI notifications config
+      */
+      if (Array.isArray(notifications)) {
+        const uniqueNotifications = _.uniqBy(
+          [...notifications].concat(newFilters.notifications),
+          function (item) {
+            return item.id;
+          }
+        );
+        newFilters.notifications = uniqueNotifications;
+      } else {
+        // check if this notification is already present if not add it
+        if (
+          newFilters.notifications.findIndex(
+            (notification: Notification) => notification.id === notifications.id
+          ) === -1
+        ) {
+          newFilters.notifications.push(notifications);
```

Comment:
(These was a previous comment about this...)

The notifications needs to be copied here. Because the newFilters returned from the reducer needs to be copied. This is passing a reference to the data the action passes in.

Reply:
fixed